packages/protobuf-test/src/next/json.test.ts

```diff
@@ -129,13 +303,34 @@ describe("json serialization", () => {
             },
           }),
         );
-        testV1Compat(Any, AnyDesc, any, {
-          typeRegistry: createRegistry(Struct, Value),
-          descSet: createDescSet(StructDesc, ValueDesc),
+        expect(
+          toJson(AnyDesc, any, {
+            descSet: createDescSet(ValueDesc, StructDesc),
+          }),
+        ).toStrictEqual({
+          "@type": "type.googleapis.com/google.protobuf.Value",
+          value: {
+            foo: 1,
+          },
+        });
+      });
+      test("json_name clash with Any.@type is not prevented", () => {
+        const any = anyPack(
+          JsonNamesMessageDesc,
+          create(JsonNamesMessageDesc, { a: "a", b: "b", c: "c" }),
+        );
+        const got = toJson(AnyDesc, any, {
+          descSet: createDescSet(JsonNamesMessageDesc),
+        });
+        expect(got).toStrictEqual({
+          "@type": "type.googleapis.com/spec.JsonNamesMessage",
+          "": "b",
+          a: "a",
         });
       });
     });
   });
+  // TODO: Add tests for remaining WKT.
```

Comment:
Sounds good to me to add JSON tests for WKT here, and keep other WKT-specific tests in `wkt` (v1 has them in `google/protobuf`).